src/hevm/src/EVM/UnitTest.hs

```diff
@@ -548,6 +550,18 @@ failOutput vm dapp UnitTestOptions { .. } testName = mconcat
       _       -> indentLines 2 (showTraceTree dapp vm)
   ]
 
+bailOutput :: Stepper.Failure -> VM -> DappInfo -> UnitTestOptions -> Text -> Text
```

Comment:
If you let `failOutput` take a `Maybe Stepper.Failure` we could merge `bailOutput` and `failOutput` and get rid of this copypasta

Reply:
We could, but I don't think we're done iterating on this area yet so imo it's a bit more flexible to keep separate until we have some bigger vision for rewriting the whole thing.